crates/ty_server/src/server/api/requests/workspace_diagnostic.rs

```diff
@@ -199,77 +198,64 @@ impl RetriableRequestHandler for WorkspaceDiagnosticRequestHandler {
 ///
 /// Diagnostics are only streamed if the client sends a partial result token.
 struct WorkspaceDiagnosticsProgressReporter<'a> {
-    total_files: usize,
-    checked_files: AtomicUsize,
     work_done: LazyWorkDoneProgress,
-    response: std::sync::Mutex<ResponseWriter<'a>>,
+    state: std::sync::Mutex<ProgressReporterState<'a>>,
```

Comment:
nit: Can we import `std::sync::Mutex` instead of fully qualifying here and below?